### PR TITLE
upgrade to bookworm

### PR DESCRIPTION
**Pull request by JudahSchwartz**
upgrade the debian image to bookworm
Must configure renegotiation for debian as well since openssl version is now >3

### CODE REVIEW

**Comment by JudahSchwartz** [comment]
I manually verified i was able to connect to this server running on my localhost

`import socket
import ssl

context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
context.load_cert_chain(certfile="server.crt", keyfile="server.key")
context.options |= 0x4

with socket.socket(socket.AF_INET6, socket.SOCK_STREAM, 0) as sock:
    sock.bind(('', 1111))
    sock.listen(5)
    with context.wrap_socket(sock, server_side=True) as ssock:
        conn, addr = ssock.accept()
        print(conn.recv(1024))
        conn.send(b"Hello, world!")`

**Comment by JudahSchwartz** [comment]
image

File: docker/python3-deb/Dockerfile [hunk 1]
```diff
@@ -28,5 +28,12 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
   python3-dev \
 && rm -rf /var/lib/apt/lists/*
 
+
+RUN cp /etc/ssl/openssl.cnf /etc/ssl/openssl.cnf.org && \
```
Comment (tkatzir) [inline docker/python3-deb/Dockerfile:32]: Maybe add a timestamp to the back filename?
```suggestion
RUN cp /etc/ssl/openssl.cnf "/etc/ssl/openssl_$(date +"%Y%m%d_%H%M").cnf" && \
```

**Review by tkatzir — APPROVED** [review]
Great! Approved.
See one suggestion.

**Comment by JudahSchwartz** [comment]
tested also against https://clienttest.ssllabs.com:8443/ssltest/viewMyClient.html

File: docker/python3-deb/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
+
 # Last modified: 2023-08-28T00:12:30.649840+00:00
 FROM python:3.10.13-slim-bullseye
```
Comment (JudahSchwartz) [inline docker/python3-deb/Dockerfile:3]:
```suggestion
FROM python:3.10.13-slim-bookworm
```

**Pull request merged**